### ISSUE TITLE
Format strings for citations

### ISSUE THREAD

**Issue by Harjot1Singh** [ShabadOS/desktop#513]
Allow format strings for customisable citations:

` `${source.englishName} Ji - ${page.english}`  `

etc etc

**Comment by bhajneet** [comment]
Would it not be easier to offer 2+ types of formattings? A dropdown with:

* MLA - `Author. "Subsection", Source, Page`
* Hyphenated `Author - Subsection - Source - Page`

And then couple it with which fields to show? Toggles for

* Author
* Subsection
* Source
* Page

This would allow BoS to choose `Format = Hyphenated, Author = True, Subsection = False, Source = True, Page = True`

**Comment by bhajneet** [comment]
Perhaps we should leave out the MLA option and just allow basics of sikhi style of citations. Would save time and we can but this on the backlog for improvements of the future

**Comment by saihaj** [comment]
I favor customizable features but they also add unnecessary complexity while writing code. I like the idea of dropdown and that can be implemented fairly quickly. So we can possible make toggles as a backlog for improvements and with this issue deliver both formats (MLA and hyphenated) in a dropdown.

**Comment by bhajneet** [comment]
I don't think many users are going to benefit from this. However we do know that the video editing team at BoS will use this feature heavily. So perhaps the default for now (until customizations are added later) should be to follow the BoS spec.

**Comment by saihaj** [comment]
> * MLA - Author. "Subsection", Source, Page
>* Hyphenated Author - Subsection - Source - Page

So these are 2 options with default being hyphenated ? If so I suggest moving this to todo and make customizations as a new issue

**Comment by bhajneet** [comment]
I don't think I explained myself correctly. Let me try again.

This issue originally planned to add a dropdown with 2 choices (MLA vs Hyphenated) and 4 toggles (Author, Subsection, Source, and Page).

From a project management perspective it looks like we're running out of time. So I suggested to put this issue on hold and also to change our default to BoS Hyphenated style. This would save us time considering we are late to deliver desktop + theme tool now. 

If we are to take that time saving route (which I believe we should), we would not be adding any options to Shabad OS for Desktop. Instead we will change the default formatting to hyphenated, and remove the "Subsection". We can open a new issue to do that super small and easy change so that it looks like this: `Author - Source - Page`

**Comment by saihaj** [comment]
Changing default format to hyphenated. Do we want that in parentheses? 
<img width="485" alt="image" src="https://user-images.githubusercontent.com/44710980/83534712-cedfd480-a4b6-11ea-8c07-ff141e9837b6.png">

**Comment by bhajneet** [comment]
Closing without resolution. This issue either affects too few users or there are too many issues with higher impact to be working on. It is possible to re-open or re-visit this issue at a later time. If you feel this issue is important, please give a thumbs-up reaction to the first post. You may track this issue from the pinned meta-tracker at the top of the issues home page.
